keps/sig-scalability/0001-enhance-network-performance-benchmark-testcases.md

```diff
@@ -0,0 +1,196 @@
+---
+title: Enhancing Network Performance Benchmarking Testcases
+authors:
+  - "@hanamantagoudvk"
+owning-sig: sig-scalability
+participating-sigs:
+- sig-networks
+- sig-testing
+reviewers:
+  - @mm4tt
+  - @aojea
+  - @wojtek-t
+approvers:
+  - TBD
+editor: 
+creation-date: 2020-08-24
+last-updated: 2020-09-17
+status: implemented
+---
+
+## Table of Contents
+
+<!-- toc -->
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-goals](#non-goals)
+  - [Challenges and Open Questions [optional]](#challenges-and-open-questions-optional)
+- [Proposal](#proposal)
+  - [Test Scenarios](#test-scenarios)
+  - [Tools](#tools)
+  - [Execution Environment](#execution-environment)
+  - [High Level Design](#high-level-design)
+  - [Resource Requirements](#resource-requirements)
+  - [Future Enhancements](#future-enhancements)
+  - [Benefits](#benefits)
+- [Graduation Criteria](#graduation-criteria)
+  - [Beta -&gt; GA](#beta---ga)
+  - [GA -&gt; Future](#ga---future)
+- [Implementation History](#implementation-history)
+- [Drawbacks [optional]](#drawbacks-optional)
+- [Alternatives [optional]](#alternatives-optional)
+<!-- /toc -->
+
+## Summary
+
+We are proposing to add more testcases for network performance benchmarking.
+
+## Motivation
+
+As Kubernetes is increasingly being adopted by various telco cloud providers for deployments, 
+it brings additional networking requirements/scenarios to be supported.  
+With wide array of network functions being containerized, the need to analyse and 
+monitor the networking performance metrics such as latency, throughput etc has arisen.
+
+We have come across 3 test frameworks which have limited testcases for network performance benchmarking.
+
+a) E2E (test\e2e\network\networking_perf.go): This framework has only 2 testcases for measuring throughput.
+
+b) netperf (https://github.com/kubernetes/perf-tests/tree/master/network/benchmarks/netperf) : This framework 
+   measures only the throughput between client-server pods for various MSS(Maximum Segment size). 
+   Other performance metrics such as latency , loss-rate , response time are not measured.
+   
+c) clusterloader2 (https://github.com/kubernetes/perf-tests/tree/master/clusterloader2) : This framework 
+   measures only network latency. Other performance metrics are not measured.
+
+### Goals
+
+The primary goal of this proposal is to enhance the testcases to cover various 
+networking scenarios :
+
+a) IPv4 , IPv6 , Dual-Stack.
+
+b) Protocols: TCP, UDP, HTTP/HTTPS.
+
+c) Pod-to-Pod , with services ClusterIP , NodePort etc.
+
+We intend to submit tests to measure the following network performance metrics for various 
+scenarios such as different services (ClusterIP/NodePort), with IPv4/IPv6 configured etc.
+
+a)  Throughput (For TCP traffic)
+
+b)  Packets per second (For UDP traffic)
+
+c)  Loss rate (For UDP traffic)
+
+d)  Latency/Jitter (For UDP/HTTP/HTTPS)
+    (Note: Latency being measured is a one-way latency)
```

Comment:
How you're going to measure this one? How you will guarantee no clock skew across machines (if there is clock skew you can't really measure it, right?)

Reply:
@wojtek-t ,  yes i understand that if there is clock skew across machines , latency metric value might be inaccurate. 
Want to know how GCP on which these scalability tests, ensures the time synchronizations ?